src/components/MoveAdder.test.ts

```diff
@@ -0,0 +1,86 @@
+import { IMove } from "battlemovr";
+import { expect } from "chai";
+import { stubBlankGame } from "../fakes.test";
+import { IPokemon } from "./Battles";
+import { MoveAdder } from "./MoveAdder";
+
+describe("MoveAdder", () => {
+
+    it("adds/replaces a new move to a Pokemon's moveset when the move index is valid", (): void => {
+        //Arrange
+        const { fsp } = stubBlankGame();
+        const pokemon: IPokemon = fsp.equations.newPokemon({
+            level: 5,
+            title: "SQUIRTLE".split(""),
+        });
+        const peck: IMove = {
+            title: "Peck",
+            remaining: 10,
+            uses: 10,
+        };
+
+        //Act
+        fsp.moveadder.addMove(pokemon, peck, 2);
```

Comment:
Here's an interesting test behavior quirk. This test is validating that when there's a valid move index, the move is added. But _how does we we know there's a valid move index?_

We happen to remember that creating a new Squirtle at level 5 gives it one existing move... but that means this test now has a behavioral dependency on that always being true. If we switch to a later generation or user scenario where it isn't, the test will fail mysteriously. It's also just unclear reading it why `2` is a valid index.

Proposal: create a `createStubPokemon` method available to all these tests that takes in a list of moves and creates a Pokemon with those moves. That way each test explicitly shows which moves already exist.